### PR TITLE
SiteIcon: Hide broken images

### LINKED ISSUE #15565: Broken Images
I keep seeing broken images in Site Results.

https://wordpress.com/read/search?q=Techcrunch&focus=1&show=Sites

screen shot 2017-06-27 at 9 47 30 am

https://wordpress.com/read/search?q=powazek&focus=1&show=Sites

screen shot 2017-06-27 at 9 49 17 am

https://wordpress.com/read/search?q=waxy&focus=1&show=Sites

screen shot 2017-06-27 at 9 49 47 am

Is there any way to avoid this? This makes the whole site feel unprofessional.

### PR DESCRIPTION
**Pull request by blowery**
Introduce a new Image component that hides itself when there's an error loading an image. Use the Image insite SiteIcon to hide broken site icons.

Fixes #15565

### CODE REVIEW

**Comment by fraying** [comment]
Tested all the URLs in #15565 and they're all fixed!

I see a brief flash of the broken image icon but it goes away quickly.

**Comment by fraying** [comment]
Can this go out? It'd be an improvement. @blowery

**Comment by samouri** [comment]
👁

**Comment by samouri** [comment]
Code LGTM.

File: client/components/image/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Image
+===============
+
+Image is a React component that wraps a normal `img` element and adds a class
```
Comment (samouri) [inline client/components/image/README.md]: maybe also specify that the class adds in a `display: none`?

**Comment by blowery** [comment]
Sorry, lost the approval in the chaos of the webpack2 revert. Will land this tomorrow.

File: client/components/image/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	componentWillReceiveProps( nextProps ) {
+		// reset the error state if we switch images
+		// TODO: support srcsets?
+		if ( nextProps.src !== this.props.src ) {
+			this.setState( { isError: false } );
+		}
+	}
+
+	handleError = () => {
+		this.setState( { isError: true } );
```
Comment (aduth) [inline client/components/image/index.js]: Should we check for existence of and preserve `onError` behavior if passed as prop?

**Pull request merged**